FilePersistence/src/version_control/merge/ListMergeComponentV2.cpp

```diff
@@ -205,4 +206,55 @@ void ListMergeComponentV2::computeOffsetsInBranch(const QList<Model::NodeIdType>
 	}
 }
 
+void ListMergeComponentV2::removeHoles(const QList<Model::NodeIdType> lists,
+													std::shared_ptr<GenericTree> tree, ChangeGraph& cg)
+{
+	for (auto listId : lists)
+	{
+		// Traverse through children labels
+		ChangeGraph::IdToLabelMap map;
+		int i = 0;
+		int label = 0;
+		int offset = 1;
+		auto listNode = tree->find(listId, true);
+		auto idList = nodeListToSortedIdList(listNode->children());
+		QHash< QString, QPair<int, int>> labelsToBeUpdated;
+		while (i < idList.size())
+		{
+			auto node = listNode->child(QString::number(label));
+			if (node)
+			{
+				map.insert(idList.at(i), {QString::number(i), MergeChange::Branch::None});
+				i++;
+				offset = 1;
+			}
+			else
+				labelsToBeUpdated.insert(QString::number(label), qMakePair(i-1, offset++));
+			label++;
+		}
+
+		auto listChanges = cg.changesForChildren(listId);
+		for (auto change : listChanges)
+		{
+			// No need to update if change is MoveOut/Deletion Or Stationary change(Not Label)
+			if (change->type() == ChangeType::Deletion ||
+				(change->type() == ChangeType::Move && change->oldParentId() == listId) ||
+				 (change->type() == ChangeType::Stationary && !change->updateFlags().testFlag(ChangeDescription::Label)))
+					continue;
+
+			// Insertion Or MoveIn or LabelChange
+			QPair<int, int> fractionalIndex;
+			if (labelsToBeUpdated.constFind(change->newLabel()) == labelsToBeUpdated.constEnd())
```

Comment:
please only call `find` once and assing the result to an `auto` value. Then you can compare this to `end` and use it in the other part of the if.